horqrux/shots.py

```diff
@@ -48,11 +48,50 @@ def finite_shots_fwd(
     """
     state = apply_gate(state, gates, values)
     n_qubits = len(state.shape)
-    mat_obs = observable_to_matrix(observable, n_qubits)
-    eigvals, eigvecs = jnp.linalg.eigh(mat_obs)
+    mat_obs = [observable_to_matrix(observable, n_qubits) for observable in observables]
```

Comment:
Would vmap work instead of some of the list comprehensions?

Reply:
I'm going to handle this one in the future, for two main reasons. For vmap to work, we need observables to be a jax Array. This raises two questions that I think we should address: 1) at the moment, we use observables as a list throughout horqrux 2) I'm not 100% sure of the jaxiness of creating arrays of observables.